OpenSim/Simulation/Model/Model.cpp

```diff
@@ -1771,26 +1774,28 @@ Vector& Model::updControls(const SimTK::State &s) const
 
 void Model::markControlsAsValid(const SimTK::State& s) const
 {
+  std::unique_lock<std::mutex> lock(*controlsCacheLock);
     if( (!_system) || (!_modelControlsIndex.isValid()) ){
-        throw Exception("Model::markControlsAsValid() requires an initialized Model./n" 
+        throw Exception("Model::markControlsAsValid() requires an initialized Model./n"
             "Prior call to Model::initSystem() is required.");
     }
 
-    Measure_<Vector>::Result controlsCache = 
+    Measure_<Vector>::Result controlsCache =
         Measure_<Vector>::Result::getAs(_system->updDefaultSubsystem()
             .getMeasure(_modelControlsIndex));
     controlsCache.markAsValid(s);
 }
 
 void Model::setControls(const SimTK::State& s, const SimTK::Vector& controls) const
-{   
+{
+  std::unique_lock<std::mutex> lock(*controlsCacheLock);
```

Comment:
> I wanted to add locks just as a precaution. Do you think this is necessary or should they just be added on a as-needed basis?

It isn't a good idea to put locks in prophylactically because they are very expensive and it leads to bad habits of throwing in locks everywhere. Better to have a complete understanding of what's going on and lock only where necessary, with an explanation of why in comments. Here I think it would be better to add a comment explaining that the current implementation is not thread safe and that all controls must be calculated in a single thread (if that is accurate).